### DISCUSSION TITLE
Can a function given by define-fun be evaluated with provided parameters and produce a result?

### DISCUSSION THREAD

**Discussion by pauljurczak** [cvc5/cvc5#9831] [Q&A] [answered]
I have a string produced by cvc5 containing a function definition:
```
(
(define-fun f ((x Int) (y Int)) Int (ite (<= 5 y) 9 x))
)
```
Is there an API, preferably Python, to evaluate this function with parameters, e.g. `f(3, 4)`, and get the result?

The function doesn't have to be represented as a string. It can be a result of `getSynthSolutions()`.

**Comment by ajreynol** [comment]
The best way to do this using the "simplify" method of our API: https://github.com/cvc5/cvc5/blob/main/include/cvc5/cvc5.h#L3895.
(Also available in Python).

You would pass the term `f(3,4)` as input (which has AST `(APPLY_UF f 3 4)`).

**Reply by pauljurczak** [reply]
How do I construct this term? Your example https://github.com/cvc5/cvc5/blob/main/examples/api/python/sygus-fun.py produces `terms = [max, min]`. How do I turn `max` into an argument for `slv.simplify()` equivalent to `max(3,4)`? 

I tried
```
slv.simplify(slv.mkTerm(Kind.APPLY_UF, max, slv.mkInteger(3), slv.mkInteger(4)))
```
but it returns `(max 3 4)` not `4`.

**Comment by ajreynol** [comment]
You should provide the implementation of `max`, i.e. the term returned as the synthesis solution for `max`, which should be a term of LAMBDA kind.

**Comment by pauljurczak** [comment, marked as answer]
Thank you. I confused inputs with outputs. This works:
```
    sol = slv.getSynthSolutions([max])
    print(slv.simplify(slv.mkTerm(Kind.APPLY_UF, sol[0], slv.mkInteger(3), slv.mkInteger(4))))
```
